File: BTArduino/app/src/main/java/com/example/btarduino/broadcast/MyBroadcastReceiver.java
package com.example.btarduino.broadcast;

import android.bluetooth.BluetoothAdapter;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.example.btarduino.initbluetooth.IBluetoothListener;

public class MyBroadcastReceiver extends BroadcastReceiver {

    private final IBluetoothListener iBluetoothListener;

    public MyBroadcastReceiver(Context context) {
        iBluetoothListener = (IBluetoothListener) context;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (action.equals(BluetoothAdapter.ACTION_STATE_CHANGED)) {
            final int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE,
                    BluetoothAdapter.ERROR);
            if (state == BluetoothAdapter.STATE_OFF) {
                iBluetoothListener.onBTSocketChange(false);
            }
        }
    }
}

File: MyHome/app/src/main/java/com/example/myhome/ui/broadcast/MyBroadCastReceiver.kt
package com.example.myhome.ui.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class MyBroadCastReceiver : BroadcastReceiver() {

    private val TAG = MyBroadCastReceiver::class.java.simpleName
    /* BluetoothAdapter.ACTION_STATE_CHANGED == intent?.action */

    override fun onReceive(context: Context?, intent: Intent?) {

    }
}